src/pages/TermsOfService.test.ts

```diff
@@ -0,0 +1,228 @@
+import { act, fireEvent, screen } from '@testing-library/react';
+import { h } from 'react-hyperscript-helpers';
+import { Ajax } from 'src/libs/ajax';
+import { Groups } from 'src/libs/ajax/Groups';
+import { Metrics } from 'src/libs/ajax/Metrics';
+import { User } from 'src/libs/ajax/User';
+import { authStore } from 'src/libs/state';
+import TermsOfServicePage from 'src/pages/TermsOfService';
+import { asMockedFn, renderWithAppContexts as render } from 'src/testing/test-utils';
+
+jest.mock('src/libs/ajax');
+jest.mock('react-notifications-component', () => {
+  return {
+    Store: {
+      addNotification: jest.fn(),
+      removeNotification: jest.fn(),
+    },
+  };
+});
+
+type NavExports = typeof import('src/libs/nav');
+jest.mock(
+  'src/libs/nav',
+  (): NavExports => ({
+    ...jest.requireActual<NavExports>('src/libs/nav'),
+    goToPath: jest.fn(),
+  })
+);
+type AuthExports = typeof import('src/libs/auth');
+jest.mock(
+  'src/libs/auth',
+  (): AuthExports => ({
+    ...jest.requireActual('src/libs/auth'),
+    signOut: jest.fn(),
+  })
+);
+
+const setupMockAjax = (termsOfService) => {
+  const getTos = jest.fn().mockResolvedValue('some text');
+  const getTermsOfServiceComplianceStatus = jest.fn().mockResolvedValue(termsOfService);
```

Comment:
`jest.fn()` will happily accept anything as its returned value/implementation. One way to document/type check the mocked value is with the `satisfies` operator.

```suggestion
  const getTermsOfServiceComplianceStatus = jest.fn().mockResolvedValue(termsOfService satisfies TermsOfServiceComplianceStatusResponse);
```

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#the-satisfies-operator